server/src/capture.rs

```diff
@@ -0,0 +1,227 @@
+// src/capture.rs
+
+// ## Imports
+//
+// ### Standard library
+use std::fs;
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+
+// ### Third-party
+use chrono::Local;
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+
+// ### Local
+
+/*
+The `Event` struct represents an event to be stored in the database.
+
+Fields:
+    - `user_id`: The ID of the user associated with the event.
+    - `event_type`: The type of event (e.g., "keystroke", "file_open").
+    - `data`: Optional additional data associated with the event.
+
+# Example
+
+let event = Event {
+    user_id: "user123".to_string(),
+    event_type: "keystroke".to_string(),
+    data: Some("Pressed key A".to_string()),
+};
+
+ */
+#[derive(Deserialize, Debug)]
+pub struct Event {
+    pub user_id: String,
+    pub event_type: String,
+    //    pub timestamp: String,
+    pub data: Option<String>,
+}
+
+/* The `Config` struct represents the database connection parameters read from `config.json`.
+
+Fields:
+    - `db_host`: The hostname or IP address of the database server.
+    - `db_user`: The username for the database connection.
+    - `db_password`: The password for the database connection.
+    - `db_name`: The name of the database.
+
+let config = Config {
+    db_host: "localhost".to_string(),
+    db_user: "your_db_user".to_string(),
+    db_password: "your_db_password".to_string(),
+    db_name: "your_db_name".to_string(),
+};
+
+*/
+#[derive(Deserialize, Serialize, Debug)]
+pub struct Config {
+    pub db_ip: String,
+    pub db_user: String,
+    pub db_password: String,
+    pub db_name: String,
+}
+
+/*
+The `EventCapture` struct provides methods to interact with the database.
+It holds a `tokio_postgres::Client` for database operations.
+
+# Usage Example
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+
+     // Create an instance of EventCapture using the configuration file
+     let event_capture = EventCapture::new("config.json").await?;
+
+     // Create an event
+     let event = Event {
+         user_id: "user123".to_string(),
+         event_type: "keystroke".to_string(),
+         data: Some("Pressed key A".to_string()),
+     };
+
+     // Insert the event into the database
+     event_capture.insert_event(event).await?;
+
+     Ok(())
+}
+ */
+pub struct EventCapture {
+    db_client: Arc<Mutex<Client>>,
+}
+
+impl EventCapture {
+    /*
+        Creates a new `EventCapture` instance by reading the database connection parameters
+        from the `config.json` file and connecting to the PostgreSQL database.
+
+        # Arguments
```

Comment:
There should be only one level-1 heading per file. Perhaps just bold it? 

**Arguments**
- `config_path`: The file path to the `config.json` file.